KitePrintSDK/src/main/java/ly/kite/KiteSDK.java

```diff
@@ -0,0 +1,320 @@
+  public static final String PAYPAL_RECIPIENT_LIVE    = "hello@kite.ly";
+
+
+  public static final String INTENT_PREFIX            = "ly.kite";
+
+
+  ////////// Static Variable(s) //////////
+
+  private static KiteSDK  sKiteSDK;
+
+
+  ////////// Member Variable(s) //////////
+
+  private Context      mApplicationContext;
+  private String       mAPIKey;
+  private Environment  mEnvironment;
+
+
+  ////////// Static Initialiser(s) //////////
+
+
+  ////////// Static Method(s) //////////
+
+  /*****************************************************
+   *
+   * Returns an instance of the SDK, assuming it has
+   * already been initialised.
+   *
+   *****************************************************/
+  public static KiteSDK getInstance( Context context )
+    {
+    if ( sKiteSDK == null )
+      {
+      // We need to create an instance, but we have only been
+      // given a context, so we need to try and load a previously
+      // saved environment.
+
+      SharedPreferences sharedPreferences = context.getSharedPreferences( SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE );
+
+
+      String apiKey = sharedPreferences.getString( SHARED_PREFERENCES_KEY_API_KEY, null );
+
+      if ( apiKey == null ) throw ( new IllegalStateException( "Unable to find persisted API key" ) );
+
+
+      String environmentName = sharedPreferences.getString( SHARED_PREFERENCES_KEY_ENVIRONMENT_NAME, null );
+
+      if ( apiKey == null ) throw ( new IllegalStateException( "Unable to find persisted environment name" ) );
+
+      try
+        {
+        Environment environment = Environment.valueOf( environmentName );
+
+        sKiteSDK = new KiteSDK( context, apiKey, environment );
+        }
+      catch ( Exception exception )
+        {
+        Log.e( LOG_TAG, "Unable to load previous environment", exception );
+        }
+
+      }
+
+    return ( sKiteSDK );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns a singleton instance of the SDK. Note that
+   * if there is already an instance of the SDK, it will
+   * have its environment set to the supplied values.
+   *
+   *****************************************************/
+  public static KiteSDK getInstance( Context context, String apiKey, Environment environment )
+    {
+    if ( sKiteSDK != null )
+      {
+      sKiteSDK.setEnvironment( apiKey, environment );
+      }
+    else
+      {
+      sKiteSDK = new KiteSDK( context, apiKey, environment );
+      }
+
+    return ( sKiteSDK );
+    }
+
+
+  /*****************************************************
+   *
+   * Convenience method for initialising and Launching the
+   * shopping experience.
+   *
+   *****************************************************/
+  public static void startFromProductSelection( Context context, String apiKey, KiteSDK.Environment environment, ArrayList<Asset> assetArrayList )
```

Comment:
hmmmm I think this (and the corresponding instance method) might be better named `startShopping` -- or perhaps something even better?. The reason being is that the journey may not actually start from Product Selection (at least when we're done) - Products can be dynamically enabled and disabled in the dashboard resulting in different start points:

* if only a single product is enabled and comes back in the template response then we can directly launch into the Product Overview stage of the journey (as there would only be one group and one product within that group)
* if there are multiple products but all those products belong to the same group i.e. "Phone Cases" we can again skip the Product Group Selection stage and jump directly to the list of products for the "Phone Cases" group.